### PR TITLE
Support enable/disable CodeLens setting for Qute templates

### PR DESCRIPTION
**Pull request by AlexXuChen**
Support enable/disable CodeLens setting for Qute templates.

Requires redhat-developer/vscode-quarkus/pull/477

Signed-off-by: Alexander Chen <alchen@redhat.com>

### CODE REVIEW

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/QuteGeneralClientSettings.java [hunk 1]
```diff
@@ -80,37 +100,56 @@ public static QuteGeneralClientSettings getGeneralQuteSettings(Object initializa
...
+	 *
 	 * @param sharedSettings the shared settings to update.
 	 * @param clientSettings the client settings used to update the shared settings.
-	 * 
+	 *
 	 * @return the state of the update of the shared settings.
 	 */
 	public static SettingsUpdateState update(SharedSettings sharedSettings, QuteGeneralClientSettings clientSettings) {
 		Map<String, QuteGeneralClientSettings> workspaceFolders = clientSettings.getWorkspaceFolders();
+
```
Comment (angelozerr) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/QuteGeneralClientSettings.java:140]: Please do like https://github.com/redhat-developer/quarkus-ls/pull/593/files#diff-397d8fbc3f031e9a83f1e1b098bbe458d76a776cfc8412f537abef5dc87202ceR153 to call cleanWorkspaceFolderSettings just one time.

**Comment by angelozerr** [comment]
The enable/disable codelens doesn't work for Java file.

Add this following line code at https://github.com/redhat-developer/quarkus-ls/blob/44759252eb62c75e1bd5be560c01c8746736aa9c/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/java/JavaFileTextDocumentService.java#L76:

```java
		if (!sharedSettings.getCodeLensSettings().isEnabled()) {
			return CompletableFuture.completedFuture(Collections.emptyList());
		}
```

like for template https://github.com/redhat-developer/quarkus-ls/blob/44759252eb62c75e1bd5be560c01c8746736aa9c/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java#L133

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/java/JavaFileTextDocumentService.java [hunk 2]
```diff
@@ -73,6 +74,9 @@ public void updateClientCapabilities(ClientCapabilities capabilities) {
 
 	@Override
 	public CompletableFuture<List<? extends CodeLens>> codeLens(CodeLensParams params) {
+		if (!sharedSettings.getCodeLensSettings().isEnabled()) {
```
Comment (AlexXuChen) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/java/JavaFileTextDocumentService.java:77]: Java file CodeLens works now

**Comment by angelozerr** [comment]
Thanks @AlexXuChen !

**Pull request merged**